crates/ruff_linter/src/rules/flake8_simplify/rules/split_static_string.rs

```diff
@@ -160,14 +165,42 @@ fn split_default(str_value: &StringLiteralValue, max_split: i32) -> Option<Expr>
             None
         }
         Ordering::Equal => {
-            let list_items: Vec<&str> = vec![str_value.to_str()];
-            Some(construct_replacement(
-                &list_items,
-                str_value.first_literal_flags(),
-            ))
+            // Behavior for maxsplit = 0 when sep is None:
+            // - If the string is empty or all whitespace, result is [].
+            // - Otherwise:
+            //   - " x ".split(maxsplit=0)  -> ['x ']
+            //   - " x ".rsplit(maxsplit=0) -> [' x']
+            //   - "".split(maxsplit=0) -> []
+            //   - " ".split(maxsplit=0) -> []
+            if string_val
+                .trim_matches(|c: char| c.is_whitespace())
+                .is_empty()
+            {
+                Some(construct_replacement(&[], str_value.first_literal_flags()))
+            } else {
+                let processed_str = if direction == Direction::Left {
+                    // Retain string from the first non-whitespace char to the end.
+                    // e.g., "  foo bar  " -> "foo bar  "
+                    let start_index = string_val.find(|c: char| !c.is_whitespace()).unwrap_or(0);
+                    &string_val[start_index..]
+                } else {
+                    // Direction::Right
+                    // Find the byte index *after* the last non-whitespace character.
+                    let end_index = string_val
+                        .rfind(|c: char| !c.is_whitespace())
+                        .map_or(string_val.len(), |i| {
+                            i + string_val[i..].chars().next().unwrap().len_utf8()
+                        });
+                    &string_val[..end_index]
+                };
```

Comment:
Can we just use the Rust `trim_start` and `trim_end` methods here:

```suggestion
                let processed_str = if direction == Direction::Left {                               
                    string_val.trim_start()                                                         
                } else {                                                                            
                    string_val.trim_end()                                                           
                };                                                     
```

Reply:
With this, I think we can flatten the whole `match` arm to something like this:

```rust
            let processed_str = if direction == Direction::Left {
                string_val.trim_start()
            } else {
                string_val.trim_end()
            };
            let list_items: &[_] = if processed_str.is_empty() {
                &[]
            } else {
                &[processed_str]
            };
            Some(construct_replacement(
                list_items,
                str_value.first_literal_flags(),
            ))
```